cli/aeon/cmd/common.go

```diff
@@ -0,0 +1,185 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/tarantool/go-tarantool/v2"
+	"github.com/tarantool/go-tlsdialer"
+	libcluster "github.com/tarantool/tt/lib/cluster"
+	libconnect "github.com/tarantool/tt/lib/connect"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// connectEtcd establishes a connection to etcd.
+func connectEtcd(uriOpts libconnect.UriOpts, connOpts connectOpts) (*clientv3.Client, error) {
+	etcdOpts := MakeEtcdOptsFromUriOpts(uriOpts)
+	if etcdOpts.Username == "" && etcdOpts.Password == "" {
+		etcdOpts.Username = connOpts.Username
+		etcdOpts.Password = connOpts.Password
+		if etcdOpts.Username == "" {
+			etcdOpts.Username = os.Getenv(libconnect.EtcdUsernameEnv)
+		}
+		if etcdOpts.Password == "" {
+			etcdOpts.Password = os.Getenv(libconnect.EtcdPasswordEnv)
+		}
+	}
+
+	etcdcli, err := libcluster.ConnectEtcd(etcdOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to etcd: %w", err)
+	}
+	return etcdcli, nil
+}
+
+// connectTarantool establishes a connection to Tarantool.
+func connectTarantool(uriOpts libconnect.UriOpts,
+	connOpts connectOpts) (tarantool.Connector, error) {
+	if uriOpts.Username == "" && uriOpts.Password == "" {
+		uriOpts.Username = connOpts.Username
+		uriOpts.Password = connOpts.Password
+		if uriOpts.Username == "" {
+			uriOpts.Username = os.Getenv(libconnect.TarantoolUsernameEnv)
+		}
+		if uriOpts.Password == "" {
+			uriOpts.Password = os.Getenv(libconnect.TarantoolPasswordEnv)
+		}
+	}
+
+	dialer, connectorOpts := MakeConnectOptsFromUriOpts(uriOpts)
+
+	ctx := context.Background()
+	if connectorOpts.Timeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, connectorOpts.Timeout)
+		defer cancel()
+	}
+	conn, err := tarantool.Connect(ctx, dialer, connectorOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to tarantool: %w", err)
+	}
+	return conn, nil
+}
+
+// doOnStorage determines a storage based on the opts.
+func doOnStorage(connOpts connectOpts, opts libconnect.UriOpts,
+	tarantoolFunc func(tarantool.Connector) error, etcdFunc func(*clientv3.Client) error) error {
+	etcdcli, errEtcd := connectEtcd(opts, connOpts)
+	if errEtcd == nil {
+		return etcdFunc(etcdcli)
+	}
+
+	conn, errTarantool := connectTarantool(opts, connOpts)
+	if errTarantool == nil {
+		return tarantoolFunc(conn)
+	}
+
+	return fmt.Errorf("failed to establish a connection to tarantool or etcd: %w, %w",
+		errTarantool, errEtcd)
+}
```

Comment:
It looks like a copy-paste actually. It would be a good idea to extract the code into some common part.

Reply:
Transfer to lib/connect?